common/src/main/java/dev/architectury/event/events/common/EntityEvent.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * This file is part of architectury.
+ * Copyright (C) 2020, 2021 architectury
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package dev.architectury.event.events.common;
+
+import dev.architectury.event.Event;
+import dev.architectury.event.EventFactory;
+import dev.architectury.event.EventResult;
+import net.minecraft.world.InteractionResultHolder;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.MobSpawnType;
+import net.minecraft.world.level.BaseSpawner;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.LevelAccessor;
+import org.jetbrains.annotations.Nullable;
+
+public interface EntityEvent {
+    /**
+     * @see LivingDeath#die(LivingEntity, DamageSource)
+     */
+    Event<LivingDeath> LIVING_DEATH = EventFactory.createEventResult();
+    /**
+     * @see LivingAttack#attack(LivingEntity, DamageSource, float)
+     */
+    Event<LivingAttack> LIVING_ATTACK = EventFactory.createEventResult();
```

Comment:
I've been contemplating whether we should change this to `LIVING_DAMAGE(D)` as that is a bit more descriptive of what's actually happening here (the living entity is taking **damage** after *being attacked*, as opposed to *attacking* something actively)

Reply:
For reference, the vanilla method is `LivingEntity#hurt`, so maybe `LIVING_HURT` might be more fitting